### ISSUE TITLE
PDO exception on version 2.6.2

### ISSUE THREAD

**Issue by daanhaitsma** [doctrine/dbal#2848]
`SQLSTATE[HY000]: General error: PDO::ATTR_STATEMENT_CLASS requires format array(classname, array(ctor_args)); the classname must be a string specifying an existing class`

I believe that in `Doctrine/DBAL/Driver/PDOConnection.php` on line 44, the class `Doctrine\DBAL\Driver\PDOStatement` cannot be found.

I am using Laravel version 5.5.2 and PHP version 7.1.8

**Comment by Ocramius** [comment]
https://github.com/doctrine/dbal/blob/8d18a33043edd1dd0b2a996262b67523e1b83f05/lib/Doctrine/DBAL/Driver/PDOStatement.php#L28 - the class is there

**Comment by daanhaitsma** [comment]
I know, I have the file as well, but the PDOConnection does not appear to be able to find it.

**Comment by Ocramius** [comment]
@daanhaitsma try producing a minimal reproducible test scenario

**Comment by JillevdW** [comment]
Any update on the status of this issue? I'm running into this exact issue.

**Comment by morozov** [comment]
@JillevdW could you provide a minimal reproducible test scenario?

**Comment by DimaUlyanets** [comment]
Had the same issue after updating composer a moment ago. Fixed that issue by rolling back:
 - Removing doctrine/common (v2.8.1)
  - Installing doctrine/common (v2.7.3) 

  - Removing doctrine/dbal (v2.7.0)
  - Installing doctrine/dbal (v2.5.13)

PHP - 7.1.5
Ubuntu - 16.04

**Comment by imonroe** [comment]
I am not sure how to trigger it reliably.  I found my way here by Googling error messages.  I was seeing on every page page that required authentication:

SQLSTATE[HY000]: General error: PDO::ATTR_STATEMENT_CLASS requires format array(classname, array(ctor_args)); the classname must be a string specifying an existing class 

It looked from the stack traces that it was being triggered because of some kind of discrepancy between the User class and the underlying users table, but beyond that I was stumped, because it seemed to work just fine locally.  FWIW, my User class had not changed, and is now working fine again with the rolled back packages.

I'm using php7.0-fpm on ubuntu 17.10 on my dev server, along with Redis as a caching driver, if that might make a difference.

**Comment by morozov** [comment]
@imonroe given you have access to the environment it's reproducible in, could you try running a  sample project like the following and see if it's sufficient?
```json
{
  "require": {
    "doctrine/common": "2.7.0",
    "doctrine/dbal": "2.5.13"
  }
}
```
```php
<?php

require __DIR__ . '/vendor/autoload.php';

$conn = \Doctrine\DBAL\DriverManager::getConnection([
    // your connection parameters
]);

$conn->query('SELECT * FROM table');
```

Until we make it reproducible, we cannot fix it.

**Comment by morozov** [comment]
Interestingly, similar non-reproducible PDO-related issues (#2628, #2847) were also reported by Laravel users. Is it possible that Laravel somehow bypasses the requirement for PDO in `composer.json` and lets you use the DBAL without the extension?

**Comment by morozov** [comment]
@imonroe please check the version of your `php-fpm`. Looks like it's lower than 7.1 but the code requires ^7.1.

**Comment by imonroe** [comment]
Upgrading PHP-FPM to 7.2 totally did the trick.

**Comment by mikeSimonson** [comment]
The person who got it working is not the one who filled the issue but it looks like this can be closed now ? @morozov

**Comment by amenk** [comment]
Had the same problem here. Downgrading to `composer  require doctrine/dbal=2.6.3` did the trick so far. Whenever; i was not able to reproduce in a minimal example on the same machine using https://github.com/doctrine/dbal/issues/2848#issuecomment-379109082 - either via php cli or with mod_php 7.1.16

**Comment by mikeSimonson** [comment]
Is the webserver that you use (the env in which you have the issue) also running php 7.1 ?

**Comment by amenk** [comment]
I just found out it is running PHP Version 7.0.29

**Comment by amenk** [comment]
I switched to php 7.1 and everything works fine with . Given the case that I composer-installed with php7.1 and then tested with php7.0 nobody can blame DBAL.

**Comment by wespiremedia** [comment]
Same problem here after `composer update`,  rolled back to 2.6.3 fixed it for now.

**Comment by digitlimit** [comment]
**Issue Resolved**
I was getting same error.
I am  running Nginx in Ubuntu 16.04, I have php7.1 installed however it turned out Nginx configuration points to php7.0-fpm.sock

![image](https://user-images.githubusercontent.com/2041419/38490245-7dbb58fe-3be0-11e8-8049-ca8527027b35.png)

Changing php7.0-fpm.sock to php7.1-fpm.sock resolved this issue

![image](https://user-images.githubusercontent.com/2041419/38490286-98b53026-3be0-11e8-913b-16c508756e9f.png)

**Comment by spernica** [comment]
We have same error because: 
PHP 7.2.2 running on server for our web app, but our **cron job** runs on PHP 7.0.29.

**Comment by Majkl578** [comment]
@spernica DBAL 2.6 does not support PHP 7.0, you will need to fix your cron job to use PHP 7.1+.
